cmd/ketch/framework_update_test.go

```diff
@@ -44,20 +60,68 @@ func Test_frameworkUpdate(t *testing.T) {
 			},
 		},
 	}
+	file, err := ioutil.TempFile("", "*.yaml")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(file.Name())
 
 	tests := []struct {
-		name    string
-		cfg     config
-		options frameworkUpdateOptions
+		name          string
+		frameworkName string
+		cfg           config
+		options       frameworkUpdateOptions
 
+		before            func()
 		wantFrameworkSpec ketchv1.FrameworkSpec
 		wantOut           string
 		wantErr           string
 	}{
 		{
-			name: "update service endpoint",
+			name:          "framework from yaml file, ignores flags",
+			frameworkName: "frontend-framework",
 			cfg: &mocks.Configuration{
-				CtrlClientObjects: []runtime.Object{frontendFramework},
+				CtrlClientObjects:    []runtime.Object{frontendFramework},
+				DynamicClientObjects: []runtime.Object{clusterIssuerStaging},
+			},
+			options: frameworkUpdateOptions{
+				name:          file.Name(),
+				appQuotaLimit: 10,
+			},
+			before: func() {
+				file.Truncate(0)
```

Comment:
looks tricky
how about `before()` returning a filename and we use it to set `options.name`. 

my point is that having functions with side effects creates additional complexity
```golang
yamlFilename := tt.createFrameworkYamlFile()
options.name = yamlFilename

err := frameworkUpdate(context.Background(), tt.cfg, tt.options, out)

```
vs
```golang
	if tt.before != nil {
		tt.before()
	}
	out := &bytes.Buffer{}
	err := frameworkUpdate(context.Background(), tt.cfg, tt.options, out)
```
something is going on there but I can't see it)

Reply:
Good call. I removed the `before()` func from all these new tests. I opted to create a temp file and write test data on each test loop. There is a little duplicated code in the `t.Run()` functions, but it's nice to be explicit about what is happening in tests.